Makefile

```diff
@@ -13,15 +23,40 @@ GOFILES=`go list ./... | grep -v "github.com/Azure/acs-engine/vendor" | sed 's|g
 
 all: build
 
-prereqs:
-	go get github.com/Masterminds/glide
-	go get github.com/jteeuwen/go-bindata/...
-	glide install
-
-build:
+.PHONE: generate
+generate:
 	go generate -v $(GOFILES)
-	go build -v -ldflags="-X github.com/Azure/acs-engine/cmd.BuildSHA=${VERSION} -X github.com/Azure/acs-engine/cmd.BuildTime=${BUILD}"
-	cd test/acs-engine-test; go build -v
+
+.PHONY: build
+build: generate
+	GOBIN=$(BINDIR) $(GO) install $(GOFLAGS) -ldflags '$(LDFLAGS)' github.com/Azure/acs-engine/cmd/...
+	cd test/acs-engine-test; go build
+
+# usage: make clean build-cross dist VERSION=v0.4.0
+.PHONY: build-cross
+build-cross: LDFLAGS += -extldflags "-static"
+build-cross:
+	CGO_ENABLED=0 gox -output="_dist/{{.OS}}-{{.Arch}}/{{.Dir}}" -osarch='$(TARGETS)' $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LDFLAGS)'
+
+.PHONY: dist
+dist:
+	( \
+		cd _dist && \
```

Comment:
Ignore me, was not running the full chain! Looks like `build-cross` creates _dist as a side effect from gox.